### PR TITLE
Refresh: Add CTA pencil banner

### PR DESCRIPTION
**Pull request by reemhamz**
## One-line summary
This pencil banner, which is to be added to all the pages that'll be using the new navigation (https://github.com/mozilla/bedrock/pull/14966), will be replacing the nav CTAs we currently have on www.mozilla.org.

This is currently a WIP 'til the new nav PR gets merged. The placements of files and imports will be updated and refined then, as well. For now, I just did what I could to get the banner showing, which will hopefully reduce whatever integration we need to get done until the new nav descends upon us ✨ 

> [!NOTE]
> Going forward with this banner without the Download Firefox alternative + logic

## Issue / Bugzilla link
Fixes https://github.com/mozilla/bedrock/issues/15095

## Testing
Make sure your your local waffle config has the following:
(P.S. new docs here https://bedrock.readthedocs.io/en/latest/install.html#testing)

```
M24_HOME = on
M24_ABOUT = on
M24_NAVIGATION_AND_FOOTER = on
```

http://localhost:8000/en-US/

### CODE REVIEW

File: bedrock/settings/base.py
Comment (alexgibson) [inline bedrock/settings/base.py:285]: **Nit:** suggest adding this to `/banners/` along with other banner string files?

File: bedrock/base/templates/includes/protocol/navigation/pencil-banner.html
Comment (alexgibson) [inline bedrock/base/templates/includes/protocol/navigation/pencil-banner.html:9]: **Issue:** this is a pretty tiny page element, so rather than it needing its own CSS bundle suggest adding this to the `m24/base-navigation-and-footer.scss` file as an import instead?

File: bedrock/base/templates/includes/protocol/navigation/pencil-banner.html
Comment (alexgibson) [inline bedrock/base/templates/includes/protocol/navigation/pencil-banner.html:11]: **Suggestion:** suggest giving this complementary landmark an `aria-label`?
```suggestion
<aside class="m24-pencil-banner c-banner" aria-label="{{ ftl('ui-promo-label') }}">
```

File: media/css/m24/pencil-banner.scss
Comment (alexgibson) [inline media/css/m24/pencil-banner.scss:11]: **Nit:** these imports might no longer be needed if you add these styles to `m24/base-navigation-and-footer.scss`?

File: media/css/m24/pencil-banner.scss
Comment (alexgibson) [inline media/css/m24/pencil-banner.scss:13]: **Issue:** on mobile the pencil banner overlaps the mobile navigation menu, making it impossible to close

image

**Review by alexgibson — CHANGES_REQUESTED** [review]
A few minor issues / suggestions but otherwise looking good. 

One other question / concern: this banner is pretty attention grabbing. If it's going to appear on all pages, are we going to add a way for people to dismiss it so they don't see it for every page navigation?

File: bedrock/base/templates/includes/protocol/navigation/pencil-banner.html
Comment (reemhamz) [reply bedrock/base/templates/includes/protocol/navigation/pencil-banner.html:9]: good suggestion!

**Comment by reemhamz** [comment]
The main reason I didn't add a dismiss button is because @stephaniehobson suggested we skip that for now with this PR. Maybe we can wait till she's back from PTO to see what she says? Or we can merge this for now and add the close function down the line?

File: bedrock/base/templates/includes/protocol/navigation/pencil-banner.html
Comment (alexgibson) [inline bedrock/base/templates/includes/protocol/navigation/pencil-banner.html:9]: Nit: you can remove this now as it's 404?

File: media/css/m24/components/navigation-refresh.scss
Comment (reemhamz) [reply media/css/m24/components/navigation-refresh.scss:15]: Hm ok so it looks like some class names got changed and that's why the styles are being finicky. I'll add a fix now

**Comment by reemhamz** [comment]
I noticed when resizing the window when the mobile nav is active, this happens:

Screenshot 2024-10-17 at 5 41 18 PM

I'm going to work on approaching this tomorrow, as I didn't have enough time today

**Comment by reemhamz** [comment]
Issue should now be fixed and is good for re-review.
This is officially my last day on the team, so maybe one of the front-enders can take this over if it needs more work.

File: media/css/m24/components/navigation-refresh.scss
Comment (alexgibson) [inline media/css/m24/components/navigation-refresh.scss:19]: **issue:** The approach here works for English, but I think we really have to try and find a way here that does not rely on magic numbers. If the banner copy flows onto three lines for example, the fix no longer works: 

image

Comment (alexgibson) [reply media/css/m24/components/navigation-refresh.scss:19]: **Suggestion:** instead of try to balance showing the banner whilst the mobile navigation menu is open, it might be simpler to just hide it instead:

https://github.com/mozilla/bedrock/blob/main/media/css/m24/components/navigation-refresh.scss#L569-L582

From a user perspective I think this makes sense too, since if someone clicks "menu" then they have shown a clear expression that they are not interested in the banner.

**Review by alexgibson — CHANGES_REQUESTED** [review]

File: media/css/m24/components/navigation-refresh.scss
Comment (alexgibson) [reply media/css/m24/components/navigation-refresh.scss]: I pushed a fix for this to the branch ^

**Review by alexgibson — APPROVED** [review]
I think the code here looks good now, but we might want to hold off merging this until https://github.com/mozilla/bedrock/issues/15315 is fixed, as content that appears above the navigation currently creates usability a usability issue when hovering over menus.

**Comment by alexgibson** [comment]
Now that https://github.com/mozilla/bedrock/pull/15361 has merged this one should hopefully be good to go. Will pull / rebase to check locally before merging.

**Comment by alexgibson** [comment]
Re-based and merging. Thanks @reemhamz 💯

**Pull request merged**
